<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Submit bcbio job | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio</title>
  <meta name="description" content="6 Submit bcbio job | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Submit bcbio job | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="labsyspharm/rnaseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Submit bcbio job | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="settings.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="extra-styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Aligning RNAseq reads with bcbio</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Quick Start</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#recommended-directory-structure"><i class="fa fa-check"></i>Recommended Directory Structure</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="prereqs.html"><a href="prereqs.html"><i class="fa fa-check"></i><b>1</b> Prerequisites in O2 environment</a></li>
<li class="chapter" data-level="2" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>2</b> Data Storage</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data.html"><a href="data.html#download-the-primary-sequence-data-to-imstor"><i class="fa fa-check"></i><b>2.1</b> Download the primary sequence data to ImStor</a></li>
<li class="chapter" data-level="2.2" data-path="data.html"><a href="data.html#copy-the-sequence-data-to-nscratch"><i class="fa fa-check"></i><b>2.2</b> Copy the sequence data to /n/scratch</a></li>
<li class="chapter" data-level="2.3" data-path="data.html"><a href="data.html#optional-convert-bcl-files-to-fastq"><i class="fa fa-check"></i><b>2.3</b> (Optional) Convert BCL files to FASTQ</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sampledesc.html"><a href="sampledesc.html"><i class="fa fa-check"></i><b>3</b> Sample Description</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sampledesc.html"><a href="sampledesc.html#digital-gene-expression"><i class="fa fa-check"></i><b>3.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="3.2" data-path="sampledesc.html"><a href="sampledesc.html#sampledesc-deep"><i class="fa fa-check"></i><b>3.2</b> Deep RNAseq</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="sampledesc.html"><a href="sampledesc.html#one-file-per-sample"><i class="fa fa-check"></i><b>3.2.1</b> One file per sample</a></li>
<li class="chapter" data-level="3.2.2" data-path="sampledesc.html"><a href="sampledesc.html#multiple-files-per-sample"><i class="fa fa-check"></i><b>3.2.2</b> Multiple files per sample</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="refgenome.html"><a href="refgenome.html"><i class="fa fa-check"></i><b>4</b> (Optional) Reference Genome</a>
<ul>
<li class="chapter" data-level="4.1" data-path="refgenome.html"><a href="refgenome.html#digital-gene-expression-1"><i class="fa fa-check"></i><b>4.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="4.2" data-path="refgenome.html"><a href="refgenome.html#spike-in"><i class="fa fa-check"></i><b>4.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="settings.html"><a href="settings.html"><i class="fa fa-check"></i><b>5</b> YAML Settings</a>
<ul>
<li class="chapter" data-level="5.1" data-path="settings.html"><a href="settings.html#digital-gene-expression-2"><i class="fa fa-check"></i><b>5.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="5.2" data-path="settings.html"><a href="settings.html#deep-rnaseq"><i class="fa fa-check"></i><b>5.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bcbio.html"><a href="bcbio.html"><i class="fa fa-check"></i><b>6</b> Submit <code>bcbio</code> job</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bcbio" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Submit <code>bcbio</code> job<a href="bcbio.html#bcbio" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>After composing the <code>O2.yaml</code> configuration file, you are ready to run <code>bcbio</code>. Instantiate the workspace by running</p>
<pre><code>bcbio_nextgen.py -w template O2.yaml alignment.csv fastq</code></pre>
<p><strong>OR</strong>, if you had <a href="sampledesc.html#sampledesc-deep">merged multiple files per sample</a>,</p>
<pre><code>bcbio_nextgen.py -w template O2.yaml alignment.csv merged</code></pre>
<p>You should now see the <code>alignment/</code> subdirectory, and the <code>work/</code> subdirectory inside of it. Descend into it:</p>
<pre><code>cd alignment/work</code></pre>
<p>All that remains to do is tell O2 the computational requirements for <code>bcbio</code>. Copy and paste the following lines into a file called <code>submit_bcbio.sh</code>:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="bcbio.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/sh</span></span>
<span id="cb28-2"><a href="bcbio.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH -p medium</span></span>
<span id="cb28-3"><a href="bcbio.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH -J bcbio_O2</span></span>
<span id="cb28-4"><a href="bcbio.html#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH -o run.o</span></span>
<span id="cb28-5"><a href="bcbio.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH -e run.e</span></span>
<span id="cb28-6"><a href="bcbio.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH -t 4-00:00</span></span>
<span id="cb28-7"><a href="bcbio.html#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --cpus-per-task=1</span></span>
<span id="cb28-8"><a href="bcbio.html#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mem=8000</span></span>
<span id="cb28-9"><a href="bcbio.html#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="bcbio.html#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span>/n/app/bcbio/tools/bin:<span class="va">$PATH</span></span>
<span id="cb28-11"><a href="bcbio.html#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="ex">bcbio_nextgen.py</span> ../config/alignment.yaml <span class="dt">\</span></span>
<span id="cb28-12"><a href="bcbio.html#cb28-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">-n</span> 24 <span class="at">-t</span> ipython <span class="at">-s</span> slurm <span class="at">-q</span> medium <span class="at">-r</span> t=4-00:00 <span class="at">--timeout</span> 2000</span></code></pre></div>
<p>More information about these settings can be found in the <a href="https://wiki.rc.hms.harvard.edu/display/O2/How+To+Submit+Parallel+Jobs+in+O2">O2 wiki</a>. Once the file is ready, use it to kick off the run (after double-checking that you are currently in the <code>alignment/work/</code> subdirectory):</p>
<pre><code>sbatch submit_bcbio.sh</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="settings.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
