<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Data Storage | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio</title>
  <meta name="description" content="2 Data Storage | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Data Storage | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="labsyspharm/rnaseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Data Storage | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="prereqs.html"/>
<link rel="next" href="sampledesc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="extra-styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Aligning RNAseq reads with bcbio</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Quick Start</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#recommended-directory-structure"><i class="fa fa-check"></i>Recommended Directory Structure</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="prereqs.html"><a href="prereqs.html"><i class="fa fa-check"></i><b>1</b> Prerequisites in O2 environment</a></li>
<li class="chapter" data-level="2" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>2</b> Data Storage</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data.html"><a href="data.html#download-the-primary-sequence-data-to-imstor"><i class="fa fa-check"></i><b>2.1</b> Download the primary sequence data to ImStor</a></li>
<li class="chapter" data-level="2.2" data-path="data.html"><a href="data.html#copy-the-sequence-data-to-nscratch"><i class="fa fa-check"></i><b>2.2</b> Copy the sequence data to /n/scratch</a></li>
<li class="chapter" data-level="2.3" data-path="data.html"><a href="data.html#optional-convert-bcl-files-to-fastq"><i class="fa fa-check"></i><b>2.3</b> (Optional) Convert BCL files to FASTQ</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sampledesc.html"><a href="sampledesc.html"><i class="fa fa-check"></i><b>3</b> Sample Description</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sampledesc.html"><a href="sampledesc.html#digital-gene-expression"><i class="fa fa-check"></i><b>3.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="3.2" data-path="sampledesc.html"><a href="sampledesc.html#sampledesc-deep"><i class="fa fa-check"></i><b>3.2</b> Deep RNAseq</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="sampledesc.html"><a href="sampledesc.html#one-file-per-sample"><i class="fa fa-check"></i><b>3.2.1</b> One file per sample</a></li>
<li class="chapter" data-level="3.2.2" data-path="sampledesc.html"><a href="sampledesc.html#multiple-files-per-sample"><i class="fa fa-check"></i><b>3.2.2</b> Multiple files per sample</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="refgenome.html"><a href="refgenome.html"><i class="fa fa-check"></i><b>4</b> (Optional) Reference Genome</a>
<ul>
<li class="chapter" data-level="4.1" data-path="refgenome.html"><a href="refgenome.html#digital-gene-expression-1"><i class="fa fa-check"></i><b>4.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="4.2" data-path="refgenome.html"><a href="refgenome.html#spike-in"><i class="fa fa-check"></i><b>4.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="settings.html"><a href="settings.html"><i class="fa fa-check"></i><b>5</b> YAML Settings</a>
<ul>
<li class="chapter" data-level="5.1" data-path="settings.html"><a href="settings.html#digital-gene-expression-2"><i class="fa fa-check"></i><b>5.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="5.2" data-path="settings.html"><a href="settings.html#deep-rnaseq"><i class="fa fa-check"></i><b>5.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bcbio.html"><a href="bcbio.html"><i class="fa fa-check"></i><b>6</b> Submit <code>bcbio</code> job</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Data Storage<a href="data.html#data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script src="./libs/ext/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<p>All of the steps in this section must be performed from the Unix system <code>transfer.rc.hms.harvard.edu</code>,
which you access via SSH using your eCommons username and password. The examples below use
the username <code>abc123</code>, so in the typed commands after the <code>$</code> you must substitute your own
username (or the username of whomever youâ€™re working on behalf of).</p>
<div id="download-the-primary-sequence-data-to-imstor" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Download the primary sequence data to ImStor<a href="data.html#download-the-primary-sequence-data-to-imstor" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>[abc123@transfer ~]$ cd /n/files/ImStor/sorger/data/rnaseq
[abc123@transfer rnaseq]$ mkdir abc123
[abc123@transfer rnaseq]$ cd abc123
[abc123@transfer abc123]$ mkdir project_name
[abc123@transfer abc123]$ cd project_name</code></pre>
<p>Now, follow the data download instructions from your sequencing facility. At HMS, the sequencing core will often share a web URL that contains your data and additional QC reports. You can download the entire directory (or a subfolder with just your FASTQ files) using a recursive <code>wget</code> command:</p>
<pre><code>[abc123@transfer project_name]$ wget -r --no-parent http://facility-url/</code></pre>
<p>replacing <code>http://facility-url/</code> with the link to the directory containing all the files you want to download. Several caveats should be kept in mind:</p>
<ul>
<li>It is important to include the trailing slash at the end of the URL provided by the sequencing core.</li>
<li>The certificate or credentials of the Bauer core file server are not always up to date. When pulling data off their server, sometimes itâ€™s necessary to include the <code>--no-check-certificate</code> flag to avoid errors. For example, the above command would look as</li>
</ul>
<pre><code>wget -r --no-parent --no-check-certificate http://facility-url/</code></pre>
</div>
<div id="copy-the-sequence-data-to-nscratch" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Copy the sequence data to /n/scratch<a href="data.html#copy-the-sequence-data-to-nscratch" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After creating a permanent copy of your data on ImStor, you will want to also create a secondary working copy on <code>/n/scratch</code>, where all <code>bcbio</code> analysis will actually be performed.</p>
<pre><code>[abc123@transfer project_name]$ cd ..
[abc123@transfer abc123]$ mkdir /n/scratch/abc123
[abc123@transfer abc123]$ rsync -avP project_name /n/scratch/abc123
[abc123@transfer abc123]$ exit</code></pre>
<p>Itâ€™s important not to put a trailing <code>/</code> on <code>project_name</code> in the <code>rsync</code> command. (If you include
the <code>/</code>, the files will end up one directory level too high)</p>
</div>
<div id="optional-convert-bcl-files-to-fastq" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> (Optional) Convert BCL files to FASTQ<a href="data.html#optional-convert-bcl-files-to-fastq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Typically, the conversion from BCL to FASTQ is done by the sequencing core. However, occasionally you may need to perform a custom conversion. To do so, you will want to use <a href="https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html">Illuminaâ€™s bcl2fastq tool</a>. The tool accepts many parameters, the precise value of which is highly dependent on the sequencer and the library prep. Consult the <a href="https://support.illumina.com/content/dam/illumina-support/documents/documentation/software_documentation/bcl2fastq/bcl2fastq2-v2-20-software-guide-15051736-03.pdf">tool guide</a> to identify the proper parameter settings that match your library preparation protocol.</p>
<p>You will then want to logout of <code>transfer.rc.hms.harvard.edu</code>, log back in to <code>o2.hms.harvard.edu</code> and request a compute node using <a href="prereqs.html#prereqs">the previously defined <code>soar</code> command</a>:</p>
<pre><code>[abc123@login01 ~]$ soar
[abc123@compute-a-01-02 ~]$</code></pre>
<p>On the compute node, navigate to your data location that contains BCL files and run the <code>bcl2fastq</code> tools:</p>
<pre><code>module load bcl2fastq
cd /n/scratch/abc123/project_name/bcl
mkdir ../fastq
bcl2fastq &lt;settings parameters&gt; --sample-sheet &lt;path/to/sample/sheet&gt; --runfolder-dir . --output-dir ../fastq</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prereqs.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sampledesc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
