<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>3 Sample Description | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio</title>
  <meta name="description" content="3 Sample Description | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="3 Sample Description | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="labsyspharm/rnaseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Sample Description | Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio" />
  
  
  

<meta name="author" content="Laboratory of Systems Pharmacology">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="data.html">
<link rel="next" href="refgenome.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="extra-styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Aligning RNAseq reads with bcbio</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Quick Start</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#recommended-directory-structure"><i class="fa fa-check"></i>Recommended Directory Structure</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="prereqs.html"><a href="prereqs.html"><i class="fa fa-check"></i><b>1</b> Prerequisites in O2 environment</a></li>
<li class="chapter" data-level="2" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>2</b> Data Storage</a><ul>
<li class="chapter" data-level="2.1" data-path="data.html"><a href="data.html#download-the-primary-sequence-data-to-imstor"><i class="fa fa-check"></i><b>2.1</b> Download the primary sequence data to ImStor</a></li>
<li class="chapter" data-level="2.2" data-path="data.html"><a href="data.html#copy-the-sequence-data-to-nscratch2"><i class="fa fa-check"></i><b>2.2</b> Copy the sequence data to /n/scratch2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sampledesc.html"><a href="sampledesc.html"><i class="fa fa-check"></i><b>3</b> Sample Description</a><ul>
<li class="chapter" data-level="3.1" data-path="sampledesc.html"><a href="sampledesc.html#digital-gene-expression"><i class="fa fa-check"></i><b>3.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="3.2" data-path="sampledesc.html"><a href="sampledesc.html#sampledesc-deep"><i class="fa fa-check"></i><b>3.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="refgenome.html"><a href="refgenome.html"><i class="fa fa-check"></i><b>4</b> Reference Genome</a><ul>
<li class="chapter" data-level="4.1" data-path="refgenome.html"><a href="refgenome.html#digital-gene-expression-1"><i class="fa fa-check"></i><b>4.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="4.2" data-path="refgenome.html"><a href="refgenome.html#spike-in"><i class="fa fa-check"></i><b>4.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="settings.html"><a href="settings.html"><i class="fa fa-check"></i><b>5</b> YAML Settings</a><ul>
<li class="chapter" data-level="5.1" data-path="settings.html"><a href="settings.html#digital-gene-expression-2"><i class="fa fa-check"></i><b>5.1</b> Digital Gene Expression</a></li>
<li class="chapter" data-level="5.2" data-path="settings.html"><a href="settings.html#deep-rnaseq"><i class="fa fa-check"></i><b>5.2</b> Deep RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bcbio.html"><a href="bcbio.html"><i class="fa fa-check"></i><b>6</b> Submit <code>bcbio</code> job</a></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Aligning deep RNAseq and Digital Gene Expression (DGE) reads using bcbio</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sampledesc" class="section level1">
<h1><span class="header-section-number">3</span> Sample Description</h1>
<p><code>bcbio</code> expects a sample description file that tells it which files map to which samples. The file should be in the .csv format and have at least two columns. An example file for four samples might look like this:</p>
<pre><code>samplename,description
S1.fastq.gz,S1
S2.fastq.gz,S2
S3.fastq.gz,S3
S4.fastq.gz,S4</code></pre>
<p>where file <code>S1.fastq.gz</code> is mapped to the sample named <code>S1</code>, and so on. Additional columns can be included in this file and get preserved by bcbio as sample metadata.</p>
<p>If you are currently in your project directory <code>/n/scratch2/abc123/myProject/</code> and have your data in the <code>fastq/</code> subdirectory, a barebones sample description file can constructed by running the following command:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">(</span><span class="bu">echo</span> <span class="st">&#39;samplename,description&#39;</span><span class="kw">;</span> <span class="kw">for</span> <span class="ex">f</span> in fastq/*fastq*<span class="kw">;</span> <span class="kw">do</span> <span class="fu">readlink</span> -f <span class="va">$f</span> <span class="kw">|</span> <span class="fu">perl</span> -pe <span class="st">&#39;s/(.*?_(S[0-9]+)_.*)/\1,\2/&#39;</span><span class="kw">;</span> <span class="kw">done)</span> <span class="op">&gt;</span> <span class="ex">alignment.csv</span></code></pre></div>
<p><strong>The rest of the guide assumes that your sample description file is named <code>alignment.csv</code>.</strong></p>
<div id="digital-gene-expression" class="section level2">
<h2><span class="header-section-number">3.1</span> Digital Gene Expression</h2>
<p>When working with DGE data, you will typically have two matching files with suffixes <code>R1</code> and <code>R2</code>. Only the <code>R1</code> file needs to be specified in the sample description file. For example, your <code>alignment.csv</code> might look like</p>
<pre><code>samplename,description
DGE1_XT_S1_R1_001.fastq.gz,MasterPlate</code></pre>
<p>where the matching <code>DGE1_XT_S1_R2_001.fastq.gz</code> is not listed explicitly.</p>
</div>
<div id="sampledesc-deep" class="section level2">
<h2><span class="header-section-number">3.2</span> Deep RNAseq</h2>
<p>In the case of deep RNAseq data, you may have multiple FASTQ files per sample. Prior to running the aligner, these files need to be merged. <code>bcbio</code> provides a script for doing so, but it needs to know which files map to which samples. Compose a file named <code>toMerge.csv</code> that provides such a mapping. For example,</p>
<pre><code>samplename,description
/n/scratch2/abc123/myProject/fastq/TRA00140445_S1_L001_R1.fastq.bz2,S1
/n/scratch2/abc123/myProject/fastq/TRA00140445_S1_L002_R1.fastq.bz2,S1
/n/scratch2/abc123/myProject/fastq/TRA00140445_S1_L003_R1.fastq.bz2,S1
/n/scratch2/abc123/myProject/fastq/TRA00140445_S1_L004_R1.fastq.bz2,S1
/n/scratch2/abc123/myProject/fastq/TRA00140445_S2_L001_R1.fastq.bz2,S2
/n/scratch2/abc123/myProject/fastq/TRA00140445_S2_L002_R1.fastq.bz2,S2
/n/scratch2/abc123/myProject/fastq/TRA00140445_S2_L003_R1.fastq.bz2,S2
/n/scratch2/abc123/myProject/fastq/TRA00140445_S2_L004_R1.fastq.bz2,S2</code></pre>
<p>lists the mapping for two samples <code>S1</code> and <code>S2</code>, each having data collected across four sequencer lanes. As above, such a file can be constructed automatically by running the following command from your project directory, while having your raw sequence files in the <code>fastq/</code> subdirectory:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">(</span><span class="bu">echo</span> <span class="st">&#39;samplename,description&#39;</span><span class="kw">;</span> <span class="kw">for</span> <span class="ex">f</span> in fastq/*fastq*<span class="kw">;</span> <span class="kw">do</span> <span class="fu">readlink</span> -f <span class="va">$f</span> <span class="kw">|</span> <span class="fu">perl</span> -pe <span class="st">&#39;s/(.*?_(S[0-9]+)_.*)/\1,\2/&#39;</span><span class="kw">;</span> <span class="kw">done)</span> <span class="op">&gt;</span> <span class="ex">toMerge.csv</span></code></pre></div>
<p>Once you have prepared the file <code>toMerge.csv</code>, run the following command to execute the merge:</p>
<pre><code>bcbio_prepare_samples.py --out merged --csv toMerge.csv</code></pre>
<p>The merging operation might take a while. When it finishes, your merged files will reside in the <code>merged/</code> subdirectory. You will also have a new filename-to-samplename mapping file <code>toMerge-merged.csv</code>. For consistency with the remainder of this guide, rename this file to <code>alignment.csv</code> using the <code>mv</code> command:</p>
<pre><code>mv toMerge-merged.csv alignment.csv</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="refgenome.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
